src/main/kotlin/org/hyacinthbots/lilybot/extensions/events/AutoThreading.kt

```diff
@@ -0,0 +1,529 @@
+			message?.channel?.asChannelOf() ?: event.getGuild().getChannelOf(proxiedMessage.channel) ?: return
+		}
+
+		val authorId: Snowflake = if (proxiedMessage == null) {
+			message?.author?.id ?: return
+		} else {
+			message?.author?.id ?: proxiedMessage.sender
+		}
+
+		val options = AutoThreadingCollection().getSingleAutoThread(channel.id) ?: return
+
+		var threadName: String? = event.message.content.trim().split("\n").firstOrNull()?.take(75)
+
+		if (!options.contentAwareNaming || threadName == null) {
+			threadName = "Thread for ${
+				message?.author?.asUser()?.username ?: proxiedMessage?.member?.name
+			}".take(75)
+		}
+
+		if (options.preventDuplicates) {
+			var previousUserThread: TextChannelThread? = null
+			val ownerThreads = ThreadsCollection().getOwnerThreads(authorId)
+
+			val threadData = ownerThreads.find { it.parentChannelId == channel.id }
+			if (threadData != null) {
+				previousUserThread = event.getGuild().getChannelOfOrNull(threadData.threadId)
+			}
+
+			if (previousUserThread != null) {
+				val response = event.message.respond {
+					content = "Please use your existing thread in this channel ${previousUserThread.mention}"
+				}
+				event.message.delete("User already has a thread")
+				response.delete(10000L, false)
+				return
+			}
+		}
+
+		val thread = channel.startPublicThreadWithMessage(
+			message?.id ?: proxiedMessage!!.channel,
+			threadName,
+			channel.data.defaultAutoArchiveDuration.value ?: ArchiveDuration.Day
+		)
+
+		ThreadsCollection().setThreadOwner(event.getGuild().id, thread.id, event.member!!.id, channel.id)
+
+		handleThreadCreation(
+			options,
+			thread,
+			message?.author ?: memberFromPk!!.asUser()
+		)
+	}
+
+	private suspend inline fun handleThreadCreation(
+		inputOptions: AutoThreadingData,
+		inputThread: TextChannelThread,
+		inputUser: User
+	) {
+		val threadMessage = inputThread.createMessage(
+			if (inputOptions.mention) {
+				inputUser.mention
+			} else {
+				"Placeholder message"
+			}
+		)
+
+		if (inputOptions.roleId != null) {
+			val role = inputThread.guild.getRole(inputOptions.roleId)
+			val moderatorRoleId = ModerationConfigCollection().getConfig(inputThread.guildId)?.role
+			var moderatorRole: Role? = null
+			if (moderatorRoleId != null) {
+				moderatorRole = inputThread.guild.getRole(moderatorRoleId)
+			}
+
+			if (moderatorRole != null && moderatorRole.mentionable) {
+				threadMessage.edit {
+					content = role.mention + moderatorRole.mention
+				}
+			} else {
+				threadMessage.edit {
+					content = role.mention
+				}
+			}
+		}
+
+		if (inputOptions.creationMessage != null) {
+			threadMessage.edit {
+				content =
+					if (inputOptions.mention) {
+						inputUser.mention + " " + inputOptions.creationMessage
+					} else {
+						inputOptions.creationMessage
+					}
+			}
+		} else {
+			threadMessage.delete("Initial thread creation")
```

Comment:
The message in the string isn't needed. As far as i can tell it's not logged anywhere, even the action log doesn't show it